server/controllers/notification.js

```diff
@@ -0,0 +1,60 @@
+const Notification = require("../models/Notification");
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+
+exports.createNotification = asyncHandler(async (req, res, next) => {
+    try {
+        const theNotification = await Notification.create(req.body);
+        res.status(200).json(theNotification);
+    } catch (err) {
+        res.status(500).json({ message: error.message });
+    }
+});
+
+exports.markAsRead = asyncHandler(async (req, res, next) => {
+    const { notificationId } = req.params;
+    try {
+        const theNotification = await Notification.findOne({ _id: notificationId });
+        theNotification.isRead = true;
+        await theNotification.save();
+        res.status(200).json(theNotification);
+    } catch (err) {
+        res.status(500).json({ message: error.message });
+    }
+});
+
+exports.getAll = asyncHandler(async (req, res, next) => {
+    try {
+        const {type, recipientId} = req.body;
+        
+        if(type !== "message"){
```

Comment:
I don't understand the reason behind this check, I think we can straight away return all the notifications for a logged in user